src/HotChocolate/Data/test/Data.EntityFramework.Tests/IntegrationTests.cs

```diff
@@ -462,4 +466,76 @@ public async Task ExecuteAsync_Should_ReturnNull_When_FirstOrDefaultZero_AsyncEn
         // assert
         result.ToJson().MatchSnapshot();
     }
+
+    [Fact]
+    public async Task ExecuteAsync_Should_ProjectCorrectly_When_Select()
+    {
+        // arrange
+        IRequestExecutor executor = await new ServiceCollection()
+            .AddGraphQL()
+            .AddFiltering()
+            .AddSorting()
+            .AddEntityFrameworkProjections()
+            .AddQueryType(
+                x => x
+                    .Name("Query")
+                    .Field("test")
+                    .Type<ObjectType<BookDto>>()
+                    .Resolve(_ => _books.AsQueryable()
+                        .Select(book =>
+                            new BookDto
+                            {
+                                Title = book.Title,
+                                Author = new AuthorDto
+                                {
+                                    Id = book.Author!.Id,
+                                    Name = book.Author.Name
+                                }
+                            })
+                    )
+                    .UseFirstOrDefault()
+                    .UseProjection()
+                    .UseFiltering()
+                    .UseSqlLogging()
```

Comment:
@PascalSenn I pulled this branch and did some debugging - this change is exactly what will resolve the EF issue.

I think your test is falsely failing - once you move this "UseSqlLogging" call before the "UseProjection" call you should get the expected sql and expression.